ui/src/store/resource.ts

```diff
@@ -286,53 +289,78 @@ export const ResourceStore = types
         self.setLoading(true);
 
         const { api } = self;
-        const json = yield api.resources();
-
-        const kinds: string[] = json.data.map((r: IResource) => r.kind);
-        kinds.forEach((k) => self.kinds.add(k));
-
-        json.data.forEach((r: IResource) => {
-          self.versions.put(r.latestVersion);
-        });
-
-        // adding the tags to the store - normalized
-        const tags: ITag[] = json.data.flatMap((item: IResource) => item.tags);
-        tags.forEach((t) => self.tags.add(t));
-
-        const allCatalogs: ICatalog[] = json.data.flatMap((item: IResource) => item.catalog);
-        allCatalogs.forEach((t) => self.catalog.put(t));
-
-        //Adding the categories to the store - normalized
-        const categories: ICategory[] = json.data.flatMap((item: IResource) => item.categories);
-        categories.forEach((c) => {
-          self.category.put(c);
-        });
 
-        const allPlatforms: IPlatform[] = json.data.flatMap((item: IResource) => item.platforms);
-        allPlatforms.forEach((p) => self.platforms.add(p));
-
-        const resources: IResource[] = json.data.map((r: IResource) => ({
-          id: r.id,
-          name: r.name,
-          resourceKey: `${r.catalog.name}/${r.kind}/${r.name}`,
-          catalog: r.catalog.id,
-          kind: r.kind,
-          latestVersion: r.latestVersion.id,
-          displayVersion: r.latestVersion.id,
-          tags: r.tags != null ? r.tags.map((tag: ITag) => tag.name) : [],
-          categories: r.categories != null ? r.categories.map((c: ICategory) => c.id) : [],
-          platforms: r.platforms != null ? r.platforms.map((p: IPlatform) => p.id) : [],
-          rating: r.rating,
-          versions: [],
-          displayName: r.latestVersion.displayName,
-          tagsString: r.tags.map((tag: ITag) => tag.name).join(' ')
-        }));
-
-        resources.forEach((r: IResource) => {
-          r.versions.push(r.latestVersion);
-          self.add(r);
-        });
+        const json = yield api.resources();
 
+        switch (true) {
+          case json === undefined:
```

Comment:
Instead of this can't we directly add a switch case on `json.status`

Reply:
in case api is not running `json` would be `undefined` itself